### ISSUE TITLE
Investigate passing a parameter through a binding 

### ISSUE THREAD

**Issue by liliankasem** [Azure/azure-functions-dotnet-worker#1080]
In the current SDK binding implementation for in-proc, users can use a trigger binding to bind to a POCO, then that to pass a parameter to the next binding.

For example in the code snippet below, we are using a BlobTrigger and binding to a `Book`, this book has an ID property. In the input binding, we are able to use `{id}` to extract the ID value from the `Book` provided by the trigger. This is then parsed by the input binding so that we can get the content of the blob named "{id}.txt". So if the blob we upload to the `book-trigger` container has the ID value set to "1". Then the input binding will grab the contents of the "1.txt" file from the `book-input` container.

```csharp
public static class BlobBookFunction
{
    [Function(nameof(BlobBookFunction))]
    public static async Task Run(
        [BlobTrigger("book-trigger", Connection = "AzureWebJobsStorage")] Book book,
        [BlobInput("book-input/{id}.txt", Connection = "AzureWebJobsStorage")] string myBlob,
        FunctionContext context)
    {
        var logger = context.GetLogger(nameof(BlobBookFunction));
        logger.LogInformation("Blob content: {content}", myBlob);
    }
}

public class Book
{
    public string Name { get; set; }
    public string Id { get; set; }
}
```

### Actions

- [x] Investigate if it is possible to make this work with out-of-proc SDK bindings
- [x] Create an issue to track the work required to enable this feature

**Comment by kshyju** [comment]
FYI. We do support using the value from one binding for another.

See this example, 
```
[Function("GetItemsFromBlob")]
public HttpResponseData Run(
     [HttpTrigger(AuthorizationLevel.Function, "get", Route ="items/{itemId}")] HttpRequestData req,
     [BlobInput("samples-workitems/{ItemId}.txt", Connection = "MyStorageConnString")] string blob
    )
{
    var response = req.CreateResponse(HttpStatusCode.OK);
    response.Headers.Add("Content-Type", "text/plain; charset=utf-8");

    response.WriteString($"Blob content: {blob}");

    return response;
}
```

This sample is reading the itemId value from the http request URL and use that for getting the blob input entry. For example if you issue a request like `/api/items/foo`, "foo" will be bound to the `ItemId` segment of the route and the blob input binding will try to get the "**foo.txt**" file from the "sample-workitems" container.

I haven't checked the child property(Id from Book) case works.

**Comment by liliankasem** [comment]
That's good to know, let me try the example you provided. I tried the book example with the reference binding object prototype we have for SDK-type bindings and got this error:

`No value for named parameter 'id'`

So something is definitely disconnected somewhere and needs investigation

![MicrosoftTeams-image](https://user-images.githubusercontent.com/2198905/192916131-0fda87f7-709f-44fe-9e4f-79c2f72df446.png)

**Comment by fabiocav** [comment]
We need to investigate whether the current binding data pipeline (in WebJobs) will continue to work once we enable SDK Type bindings.

**Comment by liliankasem** [comment]
- We found a [bug with DefaultTypes](https://github.com/Azure/azure-webjobs-sdk/issues/2930) that makes this specific scenario not work - this is a breaking change that may need to be addressed with 5.0
- We opted to make ParameterBindingData modal, meaning if an input binding uses an expression, the trigger binding will not use ParameterBindingData (the input binding will continue to use ParameterBindingData)
